working/0723-static-extensions/feature-specification-variant2.md

```diff
@@ -0,0 +1,347 @@
+way we can invoke a static member of an extension, so it may be useful for
+backward compatibility reasons.*
+
+A static member invocation on a class `C`, of the form `C.m()` (or any
+other member access), is resolved by looking up static members in `C` named
+`m` and looking up static members of every accessible extension with
+on-class `C` and a static member named `m`.
+
+If `C` contains such a declaration then the expression is an invocation of
+that static member of `C`, with the same static analysis and dynamic
+behavior as before the introduction of this feature.
+
+Otherwise, an error occurs if fewer than one or more than one declaration
+named `m` was found. *They would necessarily be declared in extensions.*
+
+Otherwise, the invocation is resolved to the given static member
+declaration in an extension named `Ej`, and the invocation is treated
+as `Ej.m()` *(this is an explicitly resolved invocation, which is specified
+above)*.
+
+#### The instantiated constructor return type of an extension
+
+We associate an extension declaration _D_ named `E` with formal type
+parameters `X1 extends B1 .. Xs extends Bs` and an actual type argument
+list `T1 .. Ts` with a type known as the _instantiated constructor return
+type of_ _D_ _with type arguments_ `T1 .. Ts`.
+
+When an extension declaration _D_ named `E` has an on-clause which is
+a non-generic class `C`, the instantiated constructor return type is `C`,
+for any list of actual type arguments.
+
+*It is not very useful to declare a type parameter of an extension
+which isn't used in the constructor return type, because it can only be
+passed in an explicitly resolved constructor invocation, e.g.,
+`E<int>.C(42)`. In all other invocations, the value of such type variables
+is determined by instantiation to bound. In any case, the type parameters
+are always ignored by static member declarations, they are only relevant to
+constructors.*
+
+When an extension declaration _D_ has no formal type parameters, and
+it has an on-type `C<S1 .. Sk>`, the instantiated constructor return type
+of _D_ is `C<S1 .. Sk>`. *In this case the on-type is a fixed type (also
+known as a ground type), e.g., `List<int>`. This implies that the
+constructor return type of D is the same for every call site.*
+
+Finally we have the general case: Consider an extension declaration
+_D_ named `E` with formal type parameters `X1 extends B1 .. Xs extends Bs`
+and a constructor return type `C<S1 .. Sk>`. With actual type arguments
+`T1 .. Ts`, the instantiated constructor return type of _D_ with type
+arguments `T1 .. Ts` is `[T1/X1 .. Ts/Xs]C<S1 .. Sk>`.
+
+#### Explicit invocation of a constructor in an extension
+
+Explicit constructor invocations are similar to static member invocations,
+but they need more detailed rules because they can use the formal type
+parameters declared by an extension.
+
+An _explicitly resolved invocation_ of a constructor named `C.name` in a
```

Comment:
The constructors are named as in the class, so that would be `C.name`. But we have the form `E.C.name` as well which is used to specify a fully explicit invocation.

Reply:
No changes made for this, that would be a separate discussion in the language team, presumably in a github issue.